src/stream/src/executor/source/source_executor.rs

```diff
@@ -175,12 +175,14 @@ impl<S: StateStore> SourceExecutor<S> {
     ) -> StreamExecutorResult<()> {
         self.metrics
             .source_split_change_count
-            .with_label_values(
+            .with_guarded_label_values(
                 &self
                     .get_metric_labels()
                     .iter()
                     .map(AsRef::as_ref)
-                    .collect::<Vec<&str>>(),
+                    .collect::<Vec<&str>>()
+                    .try_into()
+                    .unwrap(),
```

Comment:
This looks ugly... How do I convert [String; N] -> [&str; N] in a more reasonable manner..

Reply:
You can try using [each_ref](https://doc.rust-lang.org/std/primitive.array.html#method.each_ref) and [map](https://doc.rust-lang.org/std/primitive.array.html#method.map)